src/components/tikzpictureview.ts

```diff
@@ -0,0 +1,370 @@
+import * as vscode from 'vscode'
+import * as path from 'path'
+import * as fs from 'fs'
+import * as fse from 'fs-extra'
+import * as child_process from 'child_process'
+import { tmpdir } from 'os'
+import { promisify } from 'util'
+
+import { Extension } from '../main'
+
+const mkdtemp = promisify(fs.mkdtemp)
+const mkdir = promisify(fs.mkdir)
+const writeFile = promisify(fs.writeFile)
+const readFile = promisify(fs.readFile)
+const removeDir = promisify(fse.remove)
+
+interface IFileTikzCollection {
+    location: string
+    tempDir: string
+    tikzPictures: IFileTikzPicture[]
+    preamble: string
+    lastChange: number
+}
+
+interface IFileTikzPicture {
+    range: vscode.Range
+    content: string
+    tempFile: string
+    lastChange: number
+}
+
+export class TikzPictureView {
+    extension: Extension
+    tikzCollections: { [filePath: string]: IFileTikzCollection } = {}
+    initalised = false
+
+    private TEMPFOLDER_NAME = 'vscode-latexworkshop'
+
+    constructor(extension: Extension) {
+        this.extension = extension
+    }
+
+    public async view(document: vscode.TextDocument, range: vscode.Range) {
+        if (!this.initalised) {
+            await this.cleanupTempDir()
+            if (!fs.existsSync(path.join(tmpdir(), this.TEMPFOLDER_NAME))) {
+                await mkdir(path.join(tmpdir(), this.TEMPFOLDER_NAME))
```

Comment:
Use `tmp` package to make a tmp dir.

Reply:
Are there any advantages with the `tmp` package? Also, I'm rather happy with how the folder structure ends up, would such a thing be replicable with the temp package?

```
(tempdir)/vscode-latexworkshop
└── tikzpreview-tikzcl-iCEV3B
    ├── preamble.fmt
    ├── preamble.tex
    ├── tikzpicture-47712117.pdf
    ├── tikzpicture-47712117.tex
    ├── tikzpicture-47717566.pdf
    └── tikzpicture-47717566.tex
└── tikzpreview-testfile2-gK7Apl
    ├── preamble.fmt
    ├── preamble.tex
    ├── tikzpicture-47721331.pdf
    └── tikzpicture-47721331.tex
```

Reply:
Ok, close this one.